incubator/hnc/hack/ci-test.sh

```diff
@@ -27,14 +27,19 @@ set -e
 export PATH=$(go env GOPATH)/bin:$PATH
 mkdir -p $(go env GOPATH)/bin
 
+hack_dir=$(dirname ${BASH_SOURCE})
```

Comment:
Any reason to do this before installing kubebuilder? I don't think it actually _depends_ on it in any way, but it's a bit simpler IMO to say that the script a) installs dependencies and b) runs all tests.

That would also eliminate the need to say `cd -` after the first check.

Reply:
Ye, I like to fail fast if the developer didn't apply `gofmt` in the commit.
But I understand it could be better to just: `make check-fmt test` after installing the dependencies.

Changing!